### PR TITLE
filter method

### PR DESCRIPTION
**Pull request by bani2016up**

### CODE REVIEW

File: a.py
Comment (D1ffic00lt) [inline a.py:1]: bruh, it's vital

File: services/scripts.py [hunk 1]
```diff
@@ -63,6 +65,7 @@ def _normalize_musicality_rows(df: pd.DataFrame) -> None:
     def load(self) -> None:
         df = pd.read_csv(self._filepath, index_col=0)
         df.dropna(subset=MUSICALITY_ROWS, inplace=True)
+        print(len(df.Album.unique()))
```
Comment (D1ffic00lt) [inline services/scripts.py]: i think it's unnecessary

File: app.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+
+if __name__ == "__main__":
+    from services import DataBase
+    from url_parser import AuthorMetadataParser
+    db = DataBase("./data/Spotify_Youtube.csv")
+    author = "Muse"
+    print(list(db.filter([lambda x: x.artist == author, lambda x: "hole" in x.track.lower()], mode=all)))
```
Comment (D1ffic00lt) [inline app.py:8]: it looks cool, but i guess that it would be hard to understand how it works

File: services/scripts.py [hunk 3]
```diff
@@ -105,6 +108,22 @@ def similar_songs(self, song: Song, count: int = 5) -> list[Song]:
...
+        if isinstance(match_, Callable):
+            match_ = [match_]
+        return_counter = 0
+        for song in self.songs:
+            if return_counter == count:
+                break
+            match = [i(song) for i in match_]
+            if mode(match):
+                return_counter += 1
+                yield song
```
Comment (D1ffic00lt) [inline services/scripts.py:124]: that's brilliant.

**Review by D1ffic00lt — CHANGES_REQUESTED** [review]
please, remove useless files and it will be fine. and i'll squash it before i push it.  

UPD: useless file*

File: app.py (hunk 2 above)
Comment (bani2016up) [reply app.py:8]: I'll spend a little time explaining how it works.

**Comment by D1ffic00lt** [comment]
the file `a.py` still exists...

**Review by D1ffic00lt — APPROVED** [review]
looks cool

**Pull request merged**